### DISCUSSION TITLE
how to fix: 'sk_X509_free’ was not declared in this scope

### DISCUSSION THREAD

**Discussion by AceLearnGit** [openssl/openssl#22788] [Q&A] [answered]
I have a project used openssl 1.1.1n before, recently，i update openssl to 3.0.9 without any code changed, when i compile my project i got the follow err:
error: ‘sk_X509_free’ was not declared in this scope
     std::unique_ptr<stack_st_X509, decltype(&sk_X509_free)>  certChain(sk_X509_new(nullptr), sk_X509_free);
how to fix this error? Please help me.
I have include the follow header files
#include <openssl/cms.h>
#include <openssl/pem.h>
#include <openssl/err.h>
#include <openssl/x509.h>
#include <openssl/ssl.h>

**Comment by t8m** [comment]
How did you obtain the openssl/x509.h file? There should be `#define sk_X509_free(sk) OPENSSL_sk_free(ossl_check_X509_sk_type(sk))` in that file.

**Reply by t8m** [reply]
Hmm... but that won't work in this context of course because this is just a macro.

**Comment by t8m** [comment, marked as answer]
You'll have to change this call so the sk_X509_free macro is wrapped in a static function which you can then pass into that call.
